### PR TITLE
{Advisor} Migrate Advisor module to track 2 SDK

### PR DESCRIPTION
**Pull request by BigCat20196**
**Description**

- In track2, update operation add `configuration_name` as a newly required parameter.
- In track2, SDK parameters are not flattened, so we need pass object parameter instead of flattened parameters to SDK.
- Fix some tests due to changes in track2.
- Run live tests and re-record tests.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Advisor

**Comment by BigCat20196** [comment]
@jsntcy please review this PR.

File: src/azure-cli/azure/cli/command_modules/advisor/_params.py [hunk 1]
```diff
@@ -52,3 +52,5 @@ def load_arguments(self, _):
                    help='Exclude from recommendation generation.')
         c.argument('include', options_list=['--include', '-i'], action='store_true',
                    help='Include in recommendation generation.', validator=validate_include_or_exclude)
+        c.argument('configuration_name', options_list=['--configuration-name'],
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/advisor/_params.py:55]: Unncessary to use `options_list` if you don't want to add abbrevation for the argument.

Comment (BigCat20196) [reply src/azure-cli/azure/cli/command_modules/advisor/_params.py]: OK, I've removed it.

**Comment by qwordy** [comment]
Live test results
https://clitestresultstac.blob.core.windows.net/20210621-094434-836575/index.html

**Review by jsntcy — APPROVED** [review]

**Pull request merged**